recipes/xbitinfo/install_julia.patch

```diff
@@ -0,0 +1,204 @@
++    return already_ran
++
++
++def import_error_string(julia_project=None):
++    s = """
++    Required dependencies are not installed or built.  Run the following code in the Python REPL:
++
++        >>> import xbitinfo
++        >>> xbitinfo.install()
++    """
++
++    if julia_project is not None:
++        s += f"""
++        Tried to activate project {julia_project} but failed."""
++
++    return s
++
++
++def _get_julia_project(julia_project):
++    if julia_project is None:
++        is_shared = True
++        julia_project = f"xbitinfo-{__version__}"
++    else:
++        is_shared = False
++        julia_project = Path(julia_project)
++    return julia_project, is_shared
++
++
++def is_julia_version_greater_eq(Main, version="1.6"):
++    """Check if Julia version is greater than specified version."""
++    return Main.eval(f'VERSION >= v"{version}"')
++
++
++def init_julia():
++    """Initialize julia binary, turning off compiled modules if needed."""
++    from julia.core import JuliaInfo, UnsupportedPythonError
++
++    try:
++        info = JuliaInfo.load(julia="julia")
++    except FileNotFoundError:
++        env_path = os.environ["PATH"]
++        raise FileNotFoundError(
++            f"Julia is not installed in your PATH. Please install Julia and add it to your PATH.\n\nCurrent PATH: {env_path}",
++        )
++
++    if not info.is_pycall_built():
++        raise ImportError(import_error_string())
++
++    Main = None
++    try:
++        from julia import Main as _Main
++
++        Main = _Main
++    except UnsupportedPythonError:
++        # Static python binary, so we turn off pre-compiled modules.
++        from julia.core import Julia
++
++        jl = Julia(compiled_modules=False)
++        from julia import Main as _Main
++
++        Main = _Main
++
++    return Main
++
++
++def _add_to_julia_project(Main, io_arg):
++    Main.bitinformation_spec = Main.PackageSpec(
++        name="BitInformation",
++        url="https://github.com/milankl/BitInformation.jl",
++        rev="v0.6.0",
++    )
++    Main.eval(f"Pkg.add(bitinformation_spec, {io_arg})")
++
++    Main.statsbase_spec = Main.PackageSpec(
++        name="StatsBase",
++        url="https://github.com/JuliaStats/StatsBase.jl",
++        rev="v0.33.21",
++    )
++    Main.eval(f"Pkg.add(statsbase_spec, {io_arg})")
++
++
++def _escape_filename(filename):
++    """Turns a file into a string representation with correctly escaped backslashes"""
++    str_repr = str(filename)
++    str_repr = str_repr.replace("\\", "\\\\")
++    return str_repr
+diff --color -Naur xbitinfo-0.0.2.orig/xbitinfo/xbitinfo.py xbitinfo-0.0.2/xbitinfo/xbitinfo.py
+--- xbitinfo-0.0.2.orig/xbitinfo/xbitinfo.py	2022-07-11 17:37:19.000000000 -0300
++++ xbitinfo-0.0.2/xbitinfo/xbitinfo.py	2022-08-29 14:16:24.923007159 -0300
+@@ -9,6 +9,11 @@
+ 
+ from . import __version__
+ 
++from .julia_helpers import install
++already_ran = False
++if not already_ran:
++    already_ran = install()
```

Comment:
This is the trick to install/activate if it was not run before. PySr does a better job on this front but my Knowledge of Julia Pkg is pretty much zero. I have no idea how to improve this. At moment I hope it this is enough.